Tools/TestWebKitAPI/Tests/WebCore/cocoa/TestGraphicsContextGLCocoa.mm

```diff
@@ -356,6 +384,35 @@ static bool hasMultipleGPUs()
     EXPECT_TRUE(memoryFootprintChangedBy(lastFootprint, footprintChange, footprintError));
 }
 
+// Tests that prepareForDisplayWithFinishedSignal submits the underlying fence.
+// E.g. no other work will be done to context after prepare, but the finished signal will
+// eventually come.
+TEST_P(AnyBackendAPITest, FinishIsSignaled)
+{
+    auto context = createDefaultTestContext({ 2048, 2048 });
+    ASSERT_NE(context, nullptr);
+    context->markContextChanged();
+    context->clearColor(0.f, 1.f, 0.f, 1.f);
+    context->clear(WebCore::GraphicsContextGL::COLOR_BUFFER_BIT);
+    std::atomic<bool> gotSignal = false;
+    std::atomic<uint32_t> signalThreadUID = 0;
+    context->prepareForDisplayWithFinishedSignal([&gotSignal, &signalThreadUID] {
+        gotSignal = true;
+        signalThreadUID = Thread::current().uid();
+    });
+    while (!gotSignal)
+        sleep(.1_s);
+    EXPECT_TRUE(gotSignal);
```

Comment:
Is it possible for `prepareForDisplayWithFinishedSignal` to timeout? Does the possibility of timeout make this a flaky test?

Reply:
The function is non-asynchronous, e.g it is just code.
The `signal` may be never invoked, which is a bug and which is then caught by the test. The test will hang, i.e. timeout then.